### PR TITLE
Implementing `switch` expressions in `rendering/`

### PR DESCRIPTION
**Pull request by nate-thegrate**
This pull request is part of the effort to solve issue #136139.

The previous [`switch` expressions PR](https://github.com/flutter/flutter/pull/143496) was comprised of many simple changes throughout `flutter/lib/src/`, but due to some more in-depth refactoring in `flutter/lib/src/rendering/`, I decided to submit the changes to this directory as a separate pull request.

There was really just one function that I changed significantly; I'll add a comment for explanation.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/rendering/sliver_padding.dart [hunk 1]
```diff
@@ -103,22 +93,18 @@ abstract class RenderSliverEdgeInsetsPadding extends RenderSliver with RenderObj
   @override
   void performLayout() {
```
Comment (nate-thegrate) [inline packages/flutter/lib/src/rendering/sliver_padding.dart:94]: This `performLayout()` function is the spot where I did some in-depth refactoring. Originally, there was a `switch` statement with an atrocious line length:

```dart
abstract class RenderSliverEdgeInsetsPadding extends RenderSliver with RenderObjectWithChildMixin<RenderSliver> {
  void performLayout() {
    switch (applyGrowthDirectionToAxisDirection(constraints.axisDirection, constraints.growthDirection)) {
      case AxisDirection.up:
        childParentData.paintOffset = Offset(resolvedPadding!.left, calculatePaintOffset(constraints, from: resolvedPadding!.bottom + childLayoutGeometry.scrollExtent, to: resolvedPadding!.bottom + childLayoutGeometry.scrollExtent + resolvedPadding!.top));
      ...
```

After taking multiple steps to clean things up, it now looks much more reasonable:

```dart
abstract class RenderSliverEdgeInsetsPadding extends RenderSliver with RenderObjectWithChildMixin<RenderSliver> {
  void performLayout() {
    final double calculatedOffset = switch (applyGrowthDirectionToAxisDirection(constraints.axisDirection, constraints.growthDirection)) {
      AxisDirection.up => paintOffset(from: resolvedPadding.bottom + scrollExtent, to: resolvedPadding.vertical + scrollExtent),
      ...
```

**Comment by goderbauer** [comment]
@Piinks can you take a look at this one since it is mostly slivers and scrolling?

**Comment by Piinks** [comment]
> @Piinks can you take a look at this one since it is mostly slivers and scrolling?

For sure!

File: packages/flutter/lib/src/rendering/sliver_padding.dart (hunk 1 above)
Comment (Piinks) [reply packages/flutter/lib/src/rendering/sliver_padding.dart:94]: Oh yes, that was quite gnarly. Thanks for cleaning it up!

File: packages/flutter/lib/src/rendering/flex.dart [hunk 2]
```diff
@@ -792,19 +780,15 @@ class RenderFlex extends RenderBox with ContainerRenderObjectMixin<RenderBox, Fl
       } else {
         final BoxConstraints innerConstraints;
         if (crossAxisAlignment == CrossAxisAlignment.stretch) {
```
[the remark this answers is not in the document]
Comment (nate-thegrate) [reply packages/flutter/lib/src/rendering/flex.dart:782]: I think that sounds great 🙂👍

I updated `_computeSizes()` to get rid of every `==` enum comparison. Really happy with how it looks now!

**Review by Piinks — APPROVED** [review]
LGTM!

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**